Automattic/jetmon
# Added TTFB to jetmon

Added Time To First Byte (TTFB) calculation in the Jetmon monitor.
#### TTFB calculation
The `ttfb` is calculated by comparing the time between sending the **HEAD** call and receiving something on the socket.

#### Testing instructions
Work done here #8 should help to bring up a docker environment, with that, a Jetpack connected site configured in the DB, and the diff below, jetmon should create a request periodically (even if server doesn't change states) and will send message to **wpcom** with `ttfb` info under the `extra` property.
```diff
diff --git a/lib/httpcheck.js b/lib/httpcheck.js
index 11f0f8b..640523b 100644
--- a/lib/httpcheck.js
+++ b/lib/httpcheck.js
 // each worker loads it's own config object
@@ -89,34 +92,26 @@ var HttpChecker = {
                else
                        server.site_status = SITE_CONFIRMED_DOWN;
 
-               var resO    = {};
-               resO.type   = JETMON_CHECK;
-               resO.status = server.site_status;
-               resO.rtt    = Math.round( rtt / 1000 );
-               resO.code   = http_code;
-               resO.ttfb   = Math.round( ttfb / 1000 );
-               server.checks.push( resO );
-               process.send( { msgtype: 'notify_status_change', server: server } );
-
-               //if ( server.site_status !=  server.oldStatus ) {
-               //      var resO    = {};
-               //      resO.type   = JETMON_CHECK;
-               //      resO.status = server.site_status;
-               //      resO.rtt    = Math.round( rtt / 1000 );
-               //      resO.code   = http_code;
-               //      server.checks.push( resO );
-
-               //      // if site is down and it has not been confirmed
-               //      if ( server.site_status == SITE_DOWN ) {
-               //              process.send( { msgtype: 'recheck', server: server } );
-               //      } else if ( SITE_CONFIRMED_DOWN != server.site_status ) {
-               //              process.send( { msgtype: 'notify_status_change', server: server } );
-               //              slogger.trace( 'status_change: ' + JSON.stringify( server ) );
-               //      } else {
-               //              process.send( { msgtype: 'notify_still_down', server: server } );
-               //              slogger.trace( 'still_down: ' + JSON.stringify( server ) );
-               //      }
-               // }
-               // }
+               if ( server.site_status !=  server.oldStatus ) {
+                       var resO    = {};
+                       resO.type   = JETMON_CHECK;
+                       resO.status = server.site_status;
+                       resO.rtt    = Math.round( rtt / 1000 );
+                       resO.code   = http_code;
+                       resO.ttfb   = Math.round( ttfb / 1000 );
+                       server.checks.push( resO );
+
+                       // if site is down and it has not been confirmed
+                       if ( server.site_status == SITE_DOWN ) {
+                               process.send( { msgtype: 'recheck', server: server } );
+                       } else if ( SITE_CONFIRMED_DOWN != server.site_status ) {
+                               process.send( { msgtype: 'notify_status_change', server: server } );
+                               slogger.trace( 'status_change: ' + JSON.stringify( server ) );
+                       } else {
+                               process.send( { msgtype: 'notify_still_down', server: server } );
+                               slogger.trace( 'still_down: ' + JSON.stringify( server ) );
+                       }
+               }
 
```
